In [63]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
% matplotlib inline
from sklearn import linear_model
from sklearn.linear_model import LinearRegression
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import PolynomialFeatures
from sklearn.cross_validation import train_test_split
from sklearn.naive_bayes import MultinomialNB

In [54]:
core_read = pd.read_csv('spambase/spambase.data',sep=',',header=None)

In [55]:
core_read.head(3)

0     1     2   3     4     5     6     7     8     9  ...    48     49  \
0  0.00  0.64  0.64   0  0.32  0.00  0.00  0.00  0.00  0.00 ...  0.00  0.000   
1  0.21  0.28  0.50   0  0.14  0.28  0.21  0.07  0.00  0.94 ...  0.00  0.132   
2  0.06  0.00  0.71   0  1.23  0.19  0.19  0.12  0.64  0.25 ...  0.01  0.143   

   50     51     52     53     54   55    56  57  
0   0  0.778  0.000  0.000  3.756   61   278   1  
1   0  0.372  0.180  0.048  5.114  101  1028   1  
2   0  0.276  0.184  0.010  9.821  485  2259   1  

[3 rows x 58 columns]

In [58]:
y = core_read.pop(57)

In [59]:
X = core_read

In [61]:
X.head(3)

0     1     2   3     4     5     6     7     8     9   ...   47    48  \
0  0.00  0.64  0.64   0  0.32  0.00  0.00  0.00  0.00  0.00  ...    0  0.00   
1  0.21  0.28  0.50   0  0.14  0.28  0.21  0.07  0.00  0.94  ...    0  0.00   
2  0.06  0.00  0.71   0  1.23  0.19  0.19  0.12  0.64  0.25  ...    0  0.01   

      49  50     51     52     53     54   55    56  
0  0.000   0  0.778  0.000  0.000  3.756   61   278  
1  0.132   0  0.372  0.180  0.048  5.114  101  1028  
2  0.143   0  0.276  0.184  0.010  9.821  485  2259  

[3 rows x 57 columns]

In [62]:
train_X, test_x, train_y, test_y = train_test_split(X, y, test_size=0.4, random_state=101)

In [64]:
classifier = MultinomialNB()
classifier.fit(train_X, train_y)

MultinomialNB(alpha=1.0, class_prior=None, fit_prior=True)

In [65]:
classifier.score(train_X, train_y)

0.79275362318840581

In [66]:
classifier.fit(test_x, test_y)

MultinomialNB(alpha=1.0, class_prior=None, fit_prior=True)

In [67]:
classifier.score(test_x, test_y)

0.78652906029331882

Observations:
The training data resulted in an Rsq = 0.793; the test data resulted in an Rsq = 0.787.  The score is close to the idealized threashold of 0.8.  The consistency between training and test scores support a reasonable model fit (not over-fit).  The results are consistent with a reasonably effective spam capture model. 